### PR TITLE
[Work in Progress] - Updates for osmosis@1.1

### PR DESCRIPTION
**Pull request by evanshortiss**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/rchipka/node-osmosis#example
- [x] Increase the version number in the header if appropriate.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

This change accounts for osmosis being a non nested export i.e the correct version is `module.exports = osmosis`, but current typings assume `exports.osmosis = osmosis`

### CODE REVIEW

**Comment by mhegazy** [comment]
@jurajkocan can you please review this change,

File: types/osmosis/index.d.ts [hunk 1]
```diff
@@ -54,7 +55,9 @@ export interface Osmosis {
      * passing string to your function
      * result data, osmosis finished
      */
-    data(callback: (param: string) => any): Osmosis;
-}
+    data(callback: (param: any) => any): Osmosis;
```
Comment (mhegazy) [inline types/osmosis/index.d.ts]: why is this `any`?

Comment (evanshortiss) [reply types/osmosis/index.d.ts]: Typical use will probably be something along the lines of this:

```ts
interface CollectedItem {
  title: string
}

osmosis
    .get('https://lists.acme.com')
    .find('.list-item')
    .set('title')
    .data((data: CollectedItem) => {})
```

The eventual shape of the data is determined by the function chain so it's not possible to have a specific type for it AFAICT.

**Comment by RyanCavanaugh** [comment]
@evanshortiss can I merge this?

File: types/osmosis/index.d.ts (hunk 1 above)
Comment (jurajkocan) [reply types/osmosis/index.d.ts]: true, thx for the fix

**Review by jurajkocan — APPROVED** [review]
Nice, thx @evanshortiss

**Review by jurajkocan — APPROVED** [review]

**Comment by jurajkocan** [comment]
@RyanCavanaugh
You can merge this PR, thx

**Pull request merged**